modules/config/variables.tf

```
@@ -0,0 +1,45 @@
+variable "name" {
+  description = "Name to set on AWS Config resources."
+  type        = string
+  default     = "default"
+  nullable    = false
+}
+
+variable "bucket" {
+  type        = string
+  description = "The name of the S3 bucket used to store the configuration history."
+  nullable    = false
+}
+
+variable "prefix" {
+  description = "The prefix for the specified S3 bucket."
+  type        = string
+  default     = ""
+  nullable    = false
+}
+
+variable "delivery_frequency" {
+  description = "The frequency with which AWS Config recurringly delivers configuration snapshots"
```

Wow, what an API...

```suggestion
  description = "The frequency with which AWS Config recurringly delivers configuration snapshots. Valid values: One_Hour, Three_Hours, Six_Hours, Twelve_Hours, TwentyFour_Hours (https://docs.aws.amazon.com/config/latest/APIReference/API_ConfigSnapshotDeliveryProperties.html)."
```

Variable validation is not needed because the provider will handle it:

https://github.com/hashicorp/terraform-provider-aws/blob/dd99384e95b43121c6d97d545c2f366a4ba34c79/internal/service/configservice/delivery_channel.go#L73